src/table/src/requests.rs

```diff
@@ -315,21 +313,6 @@ impl TruncateTableRequest {
     }
 }
 
-pub fn valid_table_option(key: &str) -> bool {
```

Comment:
The previous validations were useful in helping users determine if their table options were correct. If we were to remove them, it is possible that some hidden bugs may appear for users. For instance, if a user accidentally types an invalid option, the database doesn't report any errors after this change, and the user may assume that their option is effective.

Reply:
The drawback is that we must enumerate all possible table options in the parser. Different engines or object stores may require different options so it's hard to provide a full list of options in the future. For example, `is_supported_in_s3()` in this function restricts a table to use S3 as storage.

Reply:
A proposal that is adding a function to the `RegionEngine` and table engine trait:

```rust
trait RegionEngine {
   fn supported_options(&self) -> Vec<&str>;
   ...
}
```

which returns the supported options for this region engine. Then the parser can use this info to check.

Reply:
I think it's ok to pass whatever the options to the region engine, as long as we can figure out whether they are valid (and throw an error if they aren't).

Reply:
Not only region engines but also object stores have different options. Adding a method to the engine traits still requires passing region engines to the parser part which makes crate dependencies complicated.

Now I check all options as a workaround.